legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperModelBuilder.java

```diff
@@ -77,6 +79,14 @@ public static org.eclipse.collections.api.block.function.Function<Property, org.
                     ._multiplicity(context.pureModel.getMultiplicity(property.multiplicity))
                     ._stereotypes(ListIterate.collect(property.stereotypes, s -> context.resolveStereotype(s.profile, s.value, s.profileSourceInformation, s.sourceInformation)))
                     ._taggedValues(ListIterate.collect(property.taggedValues, t -> new Root_meta_pure_metamodel_extension_TaggedValue_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::extension::TaggedValue"))._tag(context.resolveTag(t.tag.profile, t.tag.value, t.tag.profileSourceInformation, t.sourceInformation))._value(t.value)))
+                    ._aggregation(AggregationKind.COMPOSITE.equals(property.aggregation)
+                            ? context.pureModel.getEnumValue(M3Paths.AggregationKind, "Composite")
+                            : AggregationKind.SHARED.equals(property.aggregation)
+                            ? context.pureModel.getEnumValue(M3Paths.AggregationKind, "Shared")
+                            : AggregationKind.NONE.equals(property.aggregation)
+                            ? context.pureModel.getEnumValue(M3Paths.AggregationKind, "None")
+                            : null
```

Comment:
If property.aggregation is a non-null and not one of the previous values, then we need to throw an exception.